commit c415eab6786676d581eefda7e0f34b62706081e4
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Tue Jun 23 23:08:33 2020 +0800

    Fix .qm file installation
    
    Use standard qmake way to handle them.

diff --git a/libpeony-qt/libpeony-qt.pro b/libpeony-qt/libpeony-qt.pro
index 04cff81..427a979 100644
--- a/libpeony-qt/libpeony-qt.pro
+++ b/libpeony-qt/libpeony-qt.pro
@@ -13,7 +13,7 @@ VERSION = 2.1.0
 TARGET = peony
 TEMPLATE = lib
 
-CONFIG += link_pkgconfig no_keywords c++11
+CONFIG += link_pkgconfig no_keywords c++11 lrelease
 PKGCONFIG += glib-2.0 gio-2.0 gio-unix-2.0 poppler-qt5 gsettings-qt
 
 DEFINES += PEONYCORE_LIBRARY
@@ -56,9 +56,5 @@ unix {
     pcfile.files = development-files/peony.pc
     INSTALLS += pcfile
 
-    translation.path = /usr/share/libpeony-qt
-    translation.commands = sh $$PWD/../translations/update-translations.sh
-    translation.files += ../translations/libpeony-qt/*.ts
-    translation.files += ../translations/libpeony-qt/*.qm
-    INSTALLS += translation
+    QM_FILES_INSTALL_PATH = /usr/share/libpeony-qt
 }
diff --git a/peony-qt-desktop/peony-qt-desktop.pro b/peony-qt-desktop/peony-qt-desktop.pro
index 1f0049e..6b8abfe 100644
--- a/peony-qt-desktop/peony-qt-desktop.pro
+++ b/peony-qt-desktop/peony-qt-desktop.pro
@@ -29,7 +29,7 @@ include(../3rd-parties/SingleApplication/singleapplication.pri)
 DEFINES += QAPPLICATION_CLASS=QApplication
 
 PKGCONFIG +=gio-2.0 glib-2.0 gio-unix-2.0 gsettings-qt
-CONFIG += c++11 link_pkgconfig no_keywords
+CONFIG += c++11 link_pkgconfig no_keywords lrelease
 
 LIBS += -L$$PWD/../libpeony-qt/ -lpeony -lX11
 
@@ -64,9 +64,7 @@ HEADERS += \
 target.path = /usr/bin
 !isEmpty(target.path): INSTALLS += target
 
-translation.path = /usr/share/peony-qt-desktop
-translation.files += ../translations/peony-qt-desktop/*
-INSTALLS += translation
+QM_FILES_INSTALL_PATH = /usr/share/peony-qt-desktop
 
 data.path = /etc/xdg/autostart
 data.files += ../data/peony-desktop.desktop
diff --git a/peony-qt-plugin-test/peony-qt-menu-plugin-mate-terminal/peony-qt-menu-plugin-mate-terminal.pro b/peony-qt-plugin-test/peony-qt-menu-plugin-mate-terminal/peony-qt-menu-plugin-mate-terminal.pro
index 4eb79ea..2e4c492 100644
--- a/peony-qt-plugin-test/peony-qt-menu-plugin-mate-terminal/peony-qt-menu-plugin-mate-terminal.pro
+++ b/peony-qt-plugin-test/peony-qt-menu-plugin-mate-terminal/peony-qt-menu-plugin-mate-terminal.pro
@@ -25,7 +25,7 @@ DEFINES += QT_DEPRECATED_WARNINGS
 include(../../plugin-iface/plugin-iface.pri)
 include(../../libpeony-qt/libpeony-qt-header.pri)
 PKGCONFIG +=gio-2.0 glib-2.0 gio-unix-2.0
-CONFIG += c++11 link_pkgconfig no_keywords
+CONFIG += c++11 link_pkgconfig no_keywords lrelease
 LIBS += -L$$PWD/../../libpeony-qt/ -lpeony-qt
 
 SOURCES += \
diff --git a/src/src.pro b/src/src.pro
index 3b88733..0c86545 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -23,7 +23,7 @@ DEFINES += QAPPLICATION_CLASS=QApplication
 
 PKGCONFIG +=gio-2.0 glib-2.0 gio-unix-2.0
 LIBS +=-lgio-2.0 -lglib-2.0 -lX11
-CONFIG += c++11 link_pkgconfig no_keywords
+CONFIG += c++11 link_pkgconfig no_keywords lrelease
 
 LIBS += -L$$PWD/../libpeony-qt/ -lpeony
 
@@ -55,10 +55,7 @@ INCLUDEPATH    += ../plugin-iface
 target.path = /usr/bin
 INSTALLS += target
 
-translation.path = /usr/share/peony-qt
-translation.files += ../translations/peony-qt/*.ts
-translation.files += ../translations/peony-qt/*.qm
-INSTALLS += translation
+QM_FILES_INSTALL_PATH = /usr/share/peony-qt
 
 data.path = /usr/share/applications
 data.files += ../data/peony.desktop
diff --git a/translations/update-translations.sh b/translations/update-translations.sh
deleted file mode 100755
index 68f9bf2..0000000
--- a/translations/update-translations.sh
+++ /dev/null
@@ -1,16 +0,0 @@
-echo "update translations..."
-lupdate $(dirname $0)/../libpeony-qt/libpeony-qt.pro
-lupdate $(dirname $0)/../src/src.pro
-lupdate $(dirname $0)/../peony-qt-desktop/peony-qt-desktop.pro
-
-echo "generate new translate resources files"
-for file in $(dirname $0)/*/*.ts
-do
-	ts="*.ts"
-	qm="*.qm"
-	target=${file%.*}.qm
-	echo ${target}
-	echo "release $file $target"
-	lrelease $file $target
-done
-echo "done"
